src/index.js

```diff
@@ -44,24 +44,19 @@ module.exports = class BlockService {
   }
 
   putStream () {
-    let ps
     if (this.isOnline()) {
-      // NOTE: This will have to change in order for bitswap
-      // to understand CID
-      ps = this._bitswap.putStream()
+      return this._bitswap.putStream()
     } else {
-      ps = this._repo.blockstore.putStream()
+      return pull(
+        pull.map((blockAndCID) => {
```

Comment:
I would really like to change this, currently we are going from having a `Block`, to this strange object, and inside bitswap back to a `Block`. Can we please make the `Block` interface sufficient so we can pass it through the whole stack?